cloudinit/config/schema.py

```diff
@@ -73,11 +88,26 @@ def validate_cloudconfig_schema(config, schema, strict=False):
     """
     try:
         from jsonschema import Draft4Validator, FormatChecker
+        from jsonschema.validators import create, extend
     except ImportError:
         logging.debug(
             'Ignoring schema validation. python-jsonschema is not present')
         return
-    validator = Draft4Validator(schema, format_checker=FormatChecker())
+
+    # Add binary type checker for cloud-init schema
+    if hasattr(Draft4Validator, 'TYPE_CHECKER'):  # jsonschema 3.0+
```

Comment:
Does the python version (major or minor or both) matter?

Reply:
I believe python version doesn't matter here as this is related to allowing jsonschema to be more permissive when reading cloud-init schema values that are declared as jsonschema type 'string'.  I have updated the comment in this section to document that intent.

Instead of only permittng py2 and py3 strings for certain string type attributes, we want to allow #cloud-config yaml to provide binary data via YAML's   "!!binary  |" syntax.